### PR TITLE
Obstacles

### PR DESCRIPTION
**Pull request by barteksad**
added constraints for elbow joint

### CODE REVIEW

File: src/constraints/constraint-builder.h [hunk 1]
```diff
@@ -14,6 +14,17 @@ using namespace constraints;
 // <lower_bounds, constraint_matrix, upper_bounds>
 using QPConstraints = std::tuple<QPVector, QPMatrix, QPVector>;
 
+namespace {
```
Comment (mateusz-sulimowicz) [inline src/constraints/constraint-builder.h]: anonymous namespace in header file does not do anything

**Review by mateusz-sulimowicz — APPROVED** [review]

File: solver-example.cpp [hunk 2]
```diff
@@ -17,60 +19,61 @@ using namespace Eigen;
...
+                       { HorizontalLine{{0, 1}, {0, 0, 0.3}} },
+                       m,
+                       &inverse_kinematics);
 
 
     for (int i = 0; i < 1; ++i) {
         auto start = std::chrono::high_resolution_clock::now();
 
-        auto [e, b1] = s.run({100, 100}, {0, 100});
+        Point start_pos_gt = toPoint<6>({0,0,0,0,0,0});
```
Comment (amwms) [inline solver-example.cpp:53]: [Style] Add spaces to be consistent with the rest of the code.

File: solver-example.cpp [hunk 3]
```diff
@@ -17,60 +19,61 @@ using namespace Eigen;
...
+        Point   end_pos_gt = toPoint<6>({-M_PI,0,0,0,0,0});
```
Comment (amwms) [inline solver-example.cpp:54]: [Style] Same as above.

File: solver-example.cpp [hunk 4]
```diff
@@ -17,60 +19,61 @@ using namespace Eigen;
...
+        auto [e, b1] = s.run({0,0,0,0,0,0}, {-M_PI,0,0,0,0,0});
```
Comment (amwms) [inline solver-example.cpp:55]: [Style] Same as above.

File: solver-example.cpp [hunk 5]
```diff
@@ -17,60 +19,61 @@ using namespace Eigen;
...
+
+        Point start_pos_found = toPoint<6>({b1[0], b1[1], b1[2], b1[3], b1[4], b1[5]});
+        auto offset = DIMS * (WAYPOINTS - 1);
+        Point end_pos_found = toPoint<6>({b1[offset + 0], b1[offset + 1], b1[offset + 2], b1[offset + 3], b1[offset + 4], b1[offset + 5]});
+        offset = DIMS * 10;
+        Point mid_pos_found = toPoint<6>({b1[offset + 0], b1[offset + 1], b1[offset + 2], b1[offset + 3], b1[offset + 4], b1[offset + 5]});
+
+        std::cout << "\n\nSummary:\n";
+        std::cout << "Ground true starting position: " << start_pos_gt << " starting position after optimization: " << start_pos_found << "\n";
+        std::cout << "Middle position after optimization: "   << mid_pos_found << "\n";
```
Comment (amwms) [inline solver-example.cpp:75]: [Style] Correct spaces (?)

File: solver-example.cpp [hunk 6]
```diff
@@ -17,60 +19,61 @@ using namespace Eigen;
...
+        std::cout << "Ground true end position: "      << end_pos_gt   << " end position after optimization: "      << end_pos_found   << "\n\n";
```
Comment (amwms) [inline solver-example.cpp:76]: [Style] Same as above

**Review by amwms — APPROVED** [review]

**Pull request merged**